src/contexts/YoutubeContext.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { createContext, useContext, useState } from 'react';
+
+interface IYoutubeContext {
+  token: string;
+  setToken: (value: string) => void;
+  fetchYoutube: <T>(url: string) => Promise<T | null>;
+}
+
+const YoutubeContext = createContext<IYoutubeContext | undefined>(undefined);
+
+interface ISpotifyProvider {
+  children: JSX.Element;
+}
+
+export function YoutubeProvider(props: ISpotifyProvider) {
+  const { children } = props;
+  const [token, setToken] = useState('');
+
+  async function fetchYoutube <T>(url: string): Promise<T | null> {
+    try {
+      const promise = await fetch(`${url}?part=snippet&mine=true`, {
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${token}`,
+        },
+      });
+      return await promise.json();
+    } catch (error) {
```

Comment:
Could you add a `console.log` with the error? So in case this fails, we could get the context